fdbserver/VersionedBTree.actor.cpp

```diff
@@ -8334,6 +9021,164 @@ TEST_CASE(":/redwood/correctness/pager/cow") {
+	if (reload) {
+		pager = new DWALPager(pageSize, pagesPerExtent, fileName, cacheSizeBytes, remapCleanupWindow, concExtentReads);
+
+		wait(success(pager->init()));
+
+		LogicalPageID extID = pager->newLastExtentID();
+		m_extentQueue.create(pager, extID, "ExtentQueue", pager->newLastQueueID(), true);
+		pager->pushExtentUsedList(m_extentQueue.queueID, extID);
+
+		state int v;
+		state ExtentQueueEntry<16> e;
+		generateRandomData(e.entry, 16);
+		state int sinceYield = 0;
+		for (v = 1; v <= numEntries; ++v) {
+			// Sometimes do a commit
+			if (currentCommitSize >= targetCommitSize) {
+				printf("currentCommitSize: %d, cumulativeCommitSize: %d, pageCacheCount: %d\n",
+				       currentCommitSize,
+				       cumulativeCommitSize,
+				       pager->getPageCacheCount());
+				wait(m_extentQueue.flush());
+				wait(pager->commit());
+				cumulativeCommitSize += currentCommitSize;
+				targetCommitSize = deterministicRandom()->randomInt(2e6, 30e6);
+				currentCommitSize = 0;
+			}
+
+			// push a random entry into the queue
+			m_extentQueue.pushBack(e);
+			currentCommitSize += 16;
+
+			// yield periodically to avoid overflowing the stack
+			if (++sinceYield >= 100) {
+				sinceYield = 0;
+				wait(yield());
+			}
+		}
+		cumulativeCommitSize += currentCommitSize;
+		printf(
+		    "Final cumulativeCommitSize: %d, pageCacheCount: %d\n", cumulativeCommitSize, pager->getPageCacheCount());
+		wait(m_extentQueue.flush());
+		extentQueueState = m_extentQueue.getState();
+		printf("Commit ExtentQueue getState(): %s\n", extentQueueState.toString().c_str());
+		pager->setMetaKey(extentQueueState.asKeyRef());
+		wait(pager->commit());
+
+		Future<Void> onClosed = pager->onClosed();
+		pager->close();
+		wait(onClosed);
+	}
+
+	printf("Reopening pager file from disk.\n");
+	pager = new DWALPager(pageSize, pagesPerExtent, fileName, cacheSizeBytes, remapCleanupWindow, concExtentReads);
+	wait(success(pager->init()));
+
+	printf("Starting ExtentQueue FastPath Recovery from Disk.\n");
+
+	// reopen the pager from disk
+	state Key meta = pager->getMetaKey();
+	memcpy(&extentQueueState, meta.begin(), meta.size());
+	extentQueueState.fromKeyRef(meta);
+	printf("Recovered ExtentQueue getState(): %s\n", extentQueueState.toString().c_str());
+	m_extentQueue.recover(pager, extentQueueState, "ExtentQueueRecovered");
+
+	state double intervalStart = timer();
+	state double start = intervalStart;
+	state Standalone<VectorRef<LogicalPageID>> extentIDs = wait(pager->getUsedExtents(m_extentQueue.queueID));
+
+	// fire read requests for all used extents
+	state int i;
+	for (i = 1; i < extentIDs.size() - 1; i++) {
+		LogicalPageID extID = extentIDs[i];
+		pager->readExtent(extID);
+	}
+
+	state PromiseStream<Standalone<VectorRef<ExtentQueueEntry<16>>>> resultStream;
+	state Future<Void> queueRecoverActor;
+	queueRecoverActor = m_extentQueue.peekAllExt(resultStream);
+	state int entriesRead = 0;
+	loop choose {
```

Comment:
There is no need for a choose here as there is only one wait, just use `wait(waitNext(futurestream))`.

Reply:
Actually, see next comment, there is another case to wait for here.